pkg/kubectl/resize.go

```diff
@@ -70,11 +93,28 @@ type ReplicationControllerResizer struct {
 	client.Interface
 }
 
+// ResizeCondition is a closure around Resize that facilitates retries via util.wait
+func ResizeCondition(r Resizer, precondition *ResizePrecondition, namespace, name string, count uint) wait.ConditionFunc {
+	return func() (bool, error) {
+		_, err := r.Resize(namespace, name, precondition, count)
+		switch err.(type) {
+		case nil:
+			return true, nil
+		case ControllerResizeError:
+			e := err.(ControllerResizeError)
```

Comment:
Suggestion-- either combine this with the if statement, or the switch err.(type) statement-- my preference is the latter but your choice.